src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,120 @@
+import * as ld from "lodash";
+
+// First Group will be the variable value. Doesn't fetch the variable names
+const prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+
+export class PrologResult {
+	private readonly rawResults: string[];
+	private readonly results: Map<string, Map<number, string|boolean>>;
```

Comment:
Does this have to be a cryptic type like ` Map<string, Map<number, string | boolean>>`? In general, unless you really need the `Map` functionality (i.e. quick access by key), it's better to create interfaces because they describe much better what the data means. E.g. your `Boolean` could be a property (that probably shouldn't be called "boolean" - what does it mean?)

Reply:
What is the `number` used for? It looks like you just use that to keep track of the order in which the `string | boolean` values are found? Why do you need a `Map<number, string | boolean>` instead of a simple array `Array<string | boolean>`?